cli/lib/kontena_cli.rb

```diff
@@ -87,6 +115,27 @@ def self.cli_root(*joinables)
       File.join(Kontena.root, 'lib/kontena/cli', *joinables)
     end
   end
+
+  def self.logger
+    return @logger if @logger
+    if log_target.respond_to?(:tty?) && log_target.tty?
+      @logger = Logger.new(log_target)
+      @logger.datetime_format = "%M%S.%2N "
+    else
+      @logger = Logger.new(log_target, 1, 65536)
+      @logger.formatter = ColorStrippingFormatter.new
+    end
+    @logger.level = ENV["DEBUG"] ? Logger::DEBUG : Logger::INFO
+    @logger.progname = 'CLI'
+    @logger
+  end
+
+  class ColorStrippingFormatter < Logger::Formatter
+    def msg2str(msg)
+      msg.gsub!(/\e+\[{1,2}[0-9;:?]+m/m, '') if msg.kind_of?(String)
```

Comment:
Mutating strings is probably best avoided... and this will probably fail with `ruby2.3 --enable-frozen-string-literal`?

```ruby
    msg = msg.gsub(....)
```